Fix Maven JAR PURL detection for packages without metadata #1836

sarafarajnasardi opened the pull request:
* Add maven.py module with enhanced JAR detection for Maven packages
* Detect Maven JARs via pom.properties files and URL pattern analysis
* Convert JAR PURLs to correct Maven format (pkg:jar → pkg:maven)
* Add comprehensive test suite covering all detection scenarios
* Update scan_codebase and inspect_packages pipelines

Fixes #1836

sarafarajnasardi committed:
Return 'jar' as package type for Maven JARs #1836

* Add maven.py module with enhanced JAR detection for Maven packages
* Detect Maven JARs via pom.properties files and URL pattern analysis
* Convert JAR PURLs to correct Maven format (pkg:jar → pkg:maven)
* Add comprehensive test suite covering all detection scenarios
* Update scan_codebase and inspect_packages pipelines

Signed-off-by: Sara Faraj <sarafarajnasardi786@gmail.com>

pombredanne reviewed (changes requested):
Also, did you get all the tests passing locally? Can you paste the test run results?

pombredanne commented on scanpipe/pipes/maven.py:
```diff
@@ -0,0 +1,572 @@
+
+from packageurl import PackageURL
+
+# Try to import existing Maven utilities for better code reuse
```
This really feels like vibe coded. Do not use conditional imports, this is bad form

pombredanne reviewed (changes requested):
So:
1. reuse the code we have and drop your duplicated code entirely
2. no conditional imports. Just do 1.
3. the core mods should be in scancode-toolkit and then imported and reused here.
4. you are likely smart. Do not vibe code or use LLMs to generate the stuff as that is wasting my precious review time. Write your own code please. I will not say that twice.

sarafarajnasardi committed:
Fix Maven JAR detection

Use toolkit functions instead of custom parsing
Simplify coordinate extraction logic

Signed-off-by: Sara Faraj <sarafarajnasardi786@gmail.com>
